util/request_source.go

```diff
@@ -93,24 +92,26 @@ func IsRequestSourceInternal(reqSrc *RequestSource) bool {
 // GetRequestSource gets the request_source field of the request.
 func (r *RequestSource) GetRequestSource() string {
 	source := SourceUnknown
-	explicitSourceType := ExplicitTypeDefault
 	origin := ExternalRequest
 	if r == nil || (len(r.RequestSourceType) == 0 && len(r.ExplicitRequestSourceType) == 0) {
 		// if r.RequestSourceType and r.ExplicitRequestSourceType are not set, it's mostly possible that r.RequestSourceInternal is not set
 		// to avoid internal requests be marked as external(default value), return unknown source here.
-		return strings.Join([]string{source, explicitSourceType}, "_")
+		return source
 	}
-
+	if r.RequestSourceInternal {
+		origin = InternalRequest
+	}
+	labelList := make([]string, 0, 3)
+	labelList = append(labelList, origin)
 	if len(r.RequestSourceType) > 0 {
 		source = r.RequestSourceType
 	}
+	labelList = append(labelList, source)
 	if len(r.ExplicitRequestSourceType) > 0 {
```

Comment:
```suggestion
	if len(r.ExplicitRequestSourceType) > 0 && r.ExplicitRequestSourceType != r.RequestSourceType {
```
NIT: Maybe it's better to shorten duplicate request source